### ISSUE TITLE
Test cases: #[trace] async fn a(mut b)

### ISSUE THREAD

**Issue by taqtiqa-mark** [tikv/minitrace-rust#112]
```rust
#[minitrace::trace("a", enter_on_poll = true)]
async fn a(mut b: String) {
    c(&mut b)
}

fn c(d: &mut String) {
    d;
}
```

gives this error, which is [resolved per this comment](https://github.com/tikv/minitrace-rust/issues/112#issuecomment-1067765123):

```rust
173 | async fn a(mut b: String) {
    |                - help: consider changing this to be mutable: `mut b`
174 |     c(&mut b)
    |       ^^^^^^ cannot borrow as mutable

error[E0596]: cannot borrow `client` as mutable, as it is not declared as mutable

```

Remove the `trace` macro and the error goes away.

For anyone caught out, until a fix lands, the workaround should have no performance impact (should be optimized by the compiler):

```rust
async fn a( b: String) {
   let mut b = b;
    c(&mut b)
}
```

Note the synchronous version appears to survive intact:

```rust
#[minitrace::trace("a")]
fn a(mut b: String) {
    c(&mut b)
}

fn c(d: &mut String) {
    d;
}
```

Test cases related to issue #81.

**Comment by andylokandy** [comment]
I believe the issue comes from this statement, although I haven't had a machine to test it now:
https://github.com/tikv/minitrace-rust/blob/ede1ec506c2fc9f391f6c37489ab1cccdae5c2bb/minitrace-macro/src/lib.rs#L268

And frankly speaking, I don't know why is this statement here 🤣 because it's all copy-and-pasted from crate [async-trait](https://github.com/dtolnay/async-trait/blob/b8eb3f6da0d4828f90816269645c9c4328942f7d/src/expand.rs#L267). I guess it's an async-trait specific constraint and we can simply remove it.

**Comment by taqtiqa-mark** [comment]
Testing now.

**Comment by taqtiqa-mark** [comment]
Closed by PR #119
